File: solutions/zig/space-age/1/space_age.zig
pub const Planet = enum {
    mercury,
    venus,
    earth,
    mars,
    jupiter,
    saturn,
    uranus,
    neptune,

    pub fn age(self: Planet, seconds: usize) f64 {
        const earth_age = @as(f64, @floatFromInt(seconds)) / 31557600;
        return switch (self) {
            .mercury => earth_age / 0.2408467,
            .venus => earth_age / 0.61519726,
            .earth => earth_age,
            .mars => earth_age / 1.8808158,
            .jupiter => earth_age / 11.862615,
            .saturn => earth_age / 29.447498,
            .uranus => earth_age / 84.016846,
            .neptune => earth_age / 164.79132,
        };
    }
};

File: solutions/go/space-age/1/space_age.go
package space

type Planet string

func Age(age float64, planet Planet) float64 {
	coeff := make(map[Planet]float64)
	coeff["Earth"] = 1
	coeff["Mercury"] = 0.2408467
	coeff["Venus"] = 0.61519726
	coeff["Mars"] = 1.8808158
	coeff["Jupiter"] = 11.862615
	coeff["Saturn"] = 29.447498
	coeff["Uranus"] = 84.016846
	coeff["Neptune"] = 164.79132
	return (age / 31557600) / coeff[planet]
}
